### ISSUE TITLE
Can't override method of constructor from another module

### ISSUE THREAD

**Issue by rico345100** [facebook/flow#4643]
I have this constructor in idbcrud.js:

```javascript
function IDBCrud(table: string) { ... }
IDBCrud.prototype.get = function(...) { ... }
IDBCrud.prototype.post = function(...) { ... }

export default IDBCrud;
```

If in the same file, I can override method without any error:

```javascript
// in idbcrud.js

const accounts = new IDBCrud('Accounts');
accounts.get = function( ... ) { ... } // override
```

But in the different file, it gives me error:
```javascript
import IDBCrud from '../idbcrud';

const accounts = new IDBCrud('Accounts');
accounts.get = function( ... ) { ... }
15: accounts.get = function(match: IParams|void, options: IQueryOptions|void): Promise<Array<Object>|Error> {
              ^^^ property `get`. Property not found in
 15: accounts.get = function(match: IParams|void, options: IQueryOptions|void): Promise<Array<Object>|Error> {
     ^^^^^^^^ new object
```

Is this bug? If not, how do I solve this? Thanks!

**Comment by asolove** [comment]
A couple thoughts. First, Flow is intentionally not focused on typing pre-`class`, prototypical inheritance code. There are a number of limitations, and this one about making changes in different files is one of them.

Secondly, if you port this code to use ES6 classes, you won't be able to override the method on an individual instance, even in the same file.

For example, this code:

```js
// @flow
class Table {
  table: string;

  constructor(table: string) {
    this.table = table;
  }
  get() {
    return this.table;
  }
}

var table = new Table("a");
table.get = table.get; // or any other function, regardless of type
```

will get you an error about the last line:

    property `get` (Covariant property `get` incompatible with contravariant use in assignment of property `get`)

Which is Flow's complex way of saying that it really does not want you writing to the `get` property because changing it could allow unsafe behavior.

To summarize, I think the answer is that overwriting methods on single instances is not something Flow is going to be able to safely let you do.

There are other approaches though! If you're consistently setting this method to a specific other value, you could make a new subclass of `IDBCrud` and override the method in whatever way you want. Or you could add some attribute to the class that is checked inside the `get` method to decide in what way to behave.

**Comment by rico345100** [comment]
Ok, I just changed all of IDBCrud code to using Class and it seems it's working.
But I have one problem, when I invoke parent's method, it gives me this error:

```javascript
// IDBCrud.js
class IDBCrud {
    ...
    get(...) { ... }
}

export default IDBCrud;
```

```javascript
// accounts.js
import IDBCrud from './IDBCrud';

class AccountsCRUD extends IDBCrud {
    ...
    get(...) {
        const accounts = super.get(...);    // passing parameters to parent's get method
                                                                 // [flow] property `get` (Property not found in Object)
        ...
}
```

Why I can't invoke parent's get method? I really need it.

**Comment by asolove** [comment]
So I think the problem must be some detail of the parameters and bodies of the `get` methods. I can use super and pass parameters just fine in [simple examples like this](https://flow.org/try/#0MYGwhgzhAEAqAWBLAdgc2gbwFDWssAtgKYBc0EALgE4qoDcWO0wA9spVQK7AUtUAU+YmQ60AlJia4KSCADohRaAF48hIg1wBfRrghgAngAlE-VFSJEKtEdXG2aaSblwWKnKsmjnL1pwGpoACIAGmDoQJlEeUVNaB0dLFBIGARaACZoIgAPCiJkABNUpCdsPUMTfjEHWmcXNw8vCE4AByIqOX1jUyCjIhAQFhCgsTiExhAraApM1WQiAHc4EtR0-iCAQQLCEYZJijVhcjsnVRnOitNRoA):

```js
class Thing {
  name: string;

  constructor(name: string) {
    this.name = name;
  }

  sayHi(greeting: string): string {
    return greeting + ", " + this.name;
  }
}

class Thing2 extends Thing {
  sayHi(): string {
    return super.sayHi("Hello,");
  }
}

let t2 = new Thing2("Adam");
let name: string = t2.sayHi();
```

**Comment by rico345100** [comment]
Oh, you are correct, it was my fault. I try to access super inside of another function, and it throws error, so I made a reference to super class's method and it works now. Many thanks!
